tools/ports/freetype.py

```diff
@@ -5,56 +5,47 @@
 
 import os
 
-TAG = 'version_1'
-HASH = '0d0b1280ba0501ad0a23cf1daa1f86821c722218b59432734d3087a89acd22aabd5c3e5e1269700dcd41e87073046e906060f167c032eb91a3ac8c5808a02783'
+TAG = '2-12-1'
+HASH = '3ef3e47752b7c3cd158c738d7e0194f1c9f97ac85c754b02be6ee0f7999c3c19050f713d1e975f5310a4689337463e7b54450ef62e02c3f09864f4c6b13740d9'
 
 
 def needed(settings):
   return settings.USE_FREETYPE
 
 
 def get(ports, settings, shared):
-  ports.fetch_project('freetype', 'https://github.com/emscripten-ports/FreeType/archive/' + TAG + '.zip', 'FreeType-' + TAG, sha512hash=HASH)
+  ports.fetch_project('freetype', 'https://github.com/freetype/freetype/archive/refs/tags/VER-' + TAG + '.zip', 'FreeType-' + TAG, sha512hash=HASH)
 
   def create(final):
-    source_path = os.path.join(ports.get_dir(), 'freetype', 'FreeType-' + TAG)
+    source_path = os.path.join(ports.get_dir(), 'freetype', 'freetype-VER-' + TAG)
     ports.write_file(os.path.join(source_path, 'include/ftconfig.h'), ftconf_h)
     ports.install_header_dir(os.path.join(source_path, 'include'),
                              target=os.path.join('freetype2'))
 
+    with open(os.path.join(source_path, 'include', 'freetype', 'config', 'ftoption.h'), 'a') as ftheader:
+      ftheader.write('#define FT_CONFIG_OPTION_SYSTEM_ZLIB')
```

Comment:
Seems like you don't need these two lines since you added FT_CONFIG_OPTION_SYSTEM_ZLIB to the config below?

Reply:
I think those might be dead lines. I tried without that, and I got a bunch of duplicate symbol errors when linking.

I'm trying out a build without the config options in the zconf file now.

wasm-ld: error: duplicate symbol: get_crc_table
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(crc32.c.o)

wasm-ld: error: duplicate symbol: crc32_z
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(crc32.c.o)

wasm-ld: error: duplicate symbol: crc32_combine_op
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(crc32.c.o)

wasm-ld: error: duplicate symbol: inflateResetKeep
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(inflate.c.o)

wasm-ld: error: duplicate symbol: inflateInit_
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(inflate.c.o)

wasm-ld: error: duplicate symbol: inflateSyncPoint
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(inflate.c.o)

wasm-ld: error: duplicate symbol: inflateUndermine
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libfreetype.a(ftgzip.c.o)
>>> defined in /local/emsdk/upstream/emscripten/cache/sysroot/lib/wasm32-emscripten/libz.a(inflate.c.o)

Reply:
Removing the lines from ftconf.h and leaving this line works.

Reply:
Whats the difference between ftconf.h and ftoption.h?   It seems unfortunate to have to patch a shipping header like this.

Reply:
ftconf.h looks like a dead link. ftoption.h is referenced a lot, but there's nothing that uses ftconf.h. I'm trying out a build without ftconf.h to see if that breaks.

Reply:
Sorry I meant, `ftconfig.h` which is the file we create above.

Reply:
I removed the ftconfig.h write and file and it built cleanly, so we should be able to remove that entirely.

Reply:
There is a lot of stuff in that existing config file.   How do we know we are not changing the configuration by removing it?  Does freetype no longer use `ftconfig.h`?

Reply:
I double-checked for ftconfig.h and that's actually used in the build, so I reverted that.

The build failure is freetype depending on zlib happening first. How do I denote a dependency?

I tried using the dependency logic referenced in tools/system_libs by adding it to freetype.py, but it didn't seem to work. I see that there's tools/deps_info.py, but that doesn't have any port->port dependencies.

Reply:
You can add `deps = ['zlib']` to this file I believe.

Reply:
Change made.